src/services/EdFi.DataManagementService.Api/Backend/DeleteRequest.cs

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+using EdFi.DataManagementService.Api.Core.Model;
+
+namespace EdFi.DataManagementService.Api.Backend;
+
+/// <summary>
+/// A delete request to a document repository
+/// </summary>
+/// <param name="DocumentUuid">The document UUID to delete</param>
+/// <param name="ResourceInfo">The ResourceInfo for the resource being deleted</param>
+/// <param name="validateNoReferencesToDocument">If true, validates there are no references to the document</param>
+/// <param name="TraceId">The request TraceId</param>
+public record DeleteRequest(
```

Comment:
@CSR2017 These request/result records are backend-independent messages for the communication between DMS core and a particular backend as implemented by IDocumentStoreRepository, which will eventually be pluggable. Ed-Fi implementations will include SQL Server and PostgreSQL, but other contributors will have a well-defined interface for other backends e.g. MongoDB.